### PR TITLE
modem-manager minor plugin fixes and updates

### PR DESCRIPTION
**Pull request by aleksander0m**
Type of pull request:
- [ X] Code fix

The MR includes some fixes to the modem-manager plugin, and some minor improvements.

### CODE REVIEW

File: plugins/modem-manager/fu-mm-device.c [hunk 1]
```diff
@@ -840,9 +874,9 @@ fu_mm_device_udev_add_port (FuMmDevice	*self,
 {
 	g_return_if_fail (FU_IS_MM_DEVICE (self));
 
-	/* cdc-wdm ports always added unless one already set */
 	if (g_str_equal (subsystem, "usbmisc") &&
-	    (self->port_qmi == NULL)) {
+	    (self->port_qmi == NULL) &&
+	    (ifnum >= 0) && (ifnum == self->port_qmi_ifnum)) {
```
Comment (hughsie) [inline plugins/modem-manager/fu-mm-device.c:879]: no inner brackets pls

File: plugins/modem-manager/fu-mm-utils.c [hunk 2]
```diff
@@ -56,8 +56,22 @@ fu_mm_utils_get_udev_port_info (GUdevDevice	 *device,
 
 	if (out_port_ifnum != NULL)
 		*out_port_ifnum = port_ifnum;
-	if (out_device_sysfs_path != NULL)
-		*out_device_sysfs_path = g_steal_pointer (&device_sysfs_path);
+	if (out_device_sysfs_path != NULL) {
+		/* if device sysfs path was set for a previous port,
+		 * make sure they are all equal */
+		if (*out_device_sysfs_path != NULL) {
+			if (g_strcmp0 (device_sysfs_path, *out_device_sysfs_path) != 0) {
```
Comment (hughsie) [inline plugins/modem-manager/fu-mm-utils.c:63]: It does feel a bit dodgy using out_device_sysfs_path like this

Comment (aleksander0m) [reply plugins/modem-manager/fu-mm-utils.c:63]: Maybe I could always return the device sysfs path as output, and do the check in the caller side?

Comment (hughsie) [reply plugins/modem-manager/fu-mm-utils.c:63]: Ya, that works too.

File: plugins/modem-manager/fu-mm-device.c (hunk 1 above)
Comment (aleksander0m) [reply plugins/modem-manager/fu-mm-device.c]: Done!

File: plugins/modem-manager/fu-mm-utils.c (hunk 2 above)
Comment (aleksander0m) [reply plugins/modem-manager/fu-mm-utils.c]: Done!

File: plugins/modem-manager/fu-mm-device.c [hunk 3]
```diff
@@ -216,13 +251,20 @@ fu_mm_device_probe_default (FuDevice *device, GError **error)
...
 		fu_mm_utils_get_port_info (self->port_at, &device_sysfs_path, &self->port_at_ifnum, NULL);
-	} else if (self->port_qmi != NULL) {
-		fu_mm_utils_get_port_info (self->port_qmi, &device_sysfs_path, NULL, NULL);
-	} else {
-		g_warn_if_reached ();
+	}
+	if (self->port_qmi != NULL) {
+		g_autofree gchar *qmi_device_sysfs_path = NULL;
+		fu_mm_utils_get_port_info (self->port_qmi, &qmi_device_sysfs_path, &self->port_qmi_ifnum, NULL);
+		if (!device_sysfs_path && qmi_device_sysfs_path) {
```
Comment (hughsie) [inline plugins/modem-manager/fu-mm-device.c:260]: we tend to use != NULL and == NULL for pointers

File: plugins/modem-manager/fu-mm-device.c [hunk 4]
```diff
@@ -130,6 +154,17 @@ fu_mm_device_probe_default (FuDevice *device, GError **error)
...
 	}
+	firmware_update_str = mm_modem_firmware_update_method_build_string_from_mask (self->update_methods);
+
+	if (!validate_firmware_update_method (self->update_methods)) {
+		g_set_error (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_NOT_SUPPORTED,
+			     "invalid firmware update combination: %s",
+			     firmware_update_str);
+		return FALSE;
```
Comment (hughsie) [inline plugins/modem-manager/fu-mm-device.c:165]: could you move this to validate_firmware_update_method() perhaps? Then it returns gboolean and GError like a lot of other things in fwupd. Ta. Then LGTM.

File: plugins/modem-manager/fu-mm-device.c (hunk 3 above)
Comment (aleksander0m) [reply plugins/modem-manager/fu-mm-device.c]: I try to keep that in my mind all the time, looks like I missed this one :D Changed now

File: plugins/modem-manager/fu-mm-device.c (hunk 4 above)
Comment (aleksander0m) [reply plugins/modem-manager/fu-mm-device.c]: That's a nice change, done now

Comment (aleksander0m) [reply plugins/modem-manager/fu-mm-device.c]: Done

File: plugins/modem-manager/fu-mm-device.c (hunk 3 above)
Comment (aleksander0m) [reply plugins/modem-manager/fu-mm-device.c]: Done

**Review by superm1 — APPROVED** [review]

**Comment by aleksander0m** [comment]
Not sure if I should be the one merging this when it gets approved, or if either of you should do it :D

**Comment by superm1** [comment]
> Not sure if I should be the one merging this when it gets approved, or if either of you should do it :D

6 eggs one hand, half dozen in the other. The important part is another set of eyeballs saw it.

**Pull request merged**